tensorflow/compiler/mlir/tensorflow/ir/tf_generated_ops.td

```diff
@@ -2461,6 +2461,40 @@ this op runs. The length of the list is returned in two cases:
   );
 }
 
+def TF_DiagOp : TF_Op<"Diag", [NoSideEffect]> {
```

Comment:
Can you add `SameOperandsAndResultElementType` trait to this op?

Reply:
Thanks for pointing out this! Updated.